src/wallet/wallet2.cpp

```diff
@@ -2771,7 +2801,7 @@ bool wallet2::is_transfer_unlocked(const transfer_details& td) const
   if(!is_tx_spendtime_unlocked(td.m_tx.unlock_time, td.m_block_height))
     return false;
 
-  if(td.m_block_height + CRYPTONOTE_DEFAULT_TX_SPENDABLE_AGE > m_blockchain.size())
+  if(td.m_block_height + CRYPTONOTE_DEFAULT_TX_SPENDABLE_AGE > m_local_bc_height)
```

Comment:
That's going to complicate things when we move the wallet cache into LMDB. Seems we need a method here, that knows about both local and remote cases.